File: frontend/src/api/billing-enhanced.ts
/**
 * Phase 8: Billing Enhanced API Client
 * 재무/청구/정산 시스템 강화 API
 */

import axios from 'axios';

const API_BASE_URL = '/api/v1/billing/enhanced';

// Get auth token
const getAuthHeaders = () => {
  const token = localStorage.getItem('access_token');
  return {
    headers: {
      Authorization: `Bearer ${token}`
    }
  };
};

// Create axios instance with baseURL and auth interceptor
const apiClient = axios.create({
  baseURL: '',  // Use empty string since we're using full paths with API_BASE_URL
  timeout: 30000,
  headers: {
    'Content-Type': 'application/json',
  },
});

// Add auth header to all requests
apiClient.interceptors.request.use(
  (config) => {
    const token = localStorage.getItem('access_token');
    if (token) {
      config.headers.Authorization = `Bearer ${token}`;
      console.log('🔐 [Billing API] Token attached:', token.substring(0, 20) + '...');
    } else {
      console.error('❌ [Billing API] No token found in localStorage');
    }
    console.log('📤 [Billing API] Request:', config.method?.toUpperCase(), config.url);
    return config;
  },
  (error) => {
    console.error('❌ [Billing API] Request error:', error);
    return Promise.reject(error);
  }
);

// Add response interceptor for better error logging
apiClient.interceptors.response.use(
  (response) => {
    console.log('✅ [Billing API] Response:', response.status, response.config.url);
    return response;
  },
  (error) => {
    if (error.response) {
      console.error('❌ [Billing API] Response error:', {
        status: error.response.status,
        url: error.config?.url,
        data: error.response.data
      });
    } else {
      console.error('❌ [Billing API] Network error:', error.message);
    }
    return Promise.reject(error);
  }
);

// ============= Types =============

export interface ChargePreviewRequest {
  client_id: number;
  dispatch_date: string; // YYYY-MM-DD
  total_distance_km: number;
  pallets: number;
  weight_kg?: number;
  vehicle_type?: string;
  is_urgent: boolean;
}

export interface ChargeBreakdown {
  base_distance_charge: number;
  base_pallet_charge: number;
  base_weight_charge: number;
  subtotal: number;
  weekend_surcharge: number;
  night_surcharge: number;
  express_surcharge: number;
  temperature_control_charge: number;
  total_surcharge: number;
  volume_discount: number;
  special_discount: number;
  total_discount: number;
  subtotal_after_discount: number;
  tax_amount: number;
  total_amount: number;
}

export interface ChargePreviewResponse {
  client_id: number;
  dispatch_date: string;
  breakdown: ChargeBreakdown;
  policy_info: Record<string, any>;
  notes: string[];
}

export interface FinancialSummary {
  period_start: string;
  period_end: string;
  total_revenue: number;
  invoiced_amount: number;
  collected_amount: number;
  collection_rate: number;
  total_receivables: number;
  current_receivables: number;
  overdue_receivables: number;
  overdue_count: number;
  total_settlements: number;
  pending_settlements: number;
  paid_settlements: number;
  cash_in: number;
  cash_out: number;
  net_cash_flow: number;
}

export interface MonthlyTrend {
  month: string; // YYYY-MM
  revenue: number;
  collected: number;
  settlements: number;
  net_profit: number;
}

export interface TopClient {
  client_id: number;
  client_name: string;
  total_revenue: number;
  invoice_count: number;
  collection_rate: number;
}

export interface AutoInvoiceSchedule {
  id: number;
  client_id: number;
  enabled: boolean;
  billing_day: number;
  auto_send_email: boolean;
  send_reminder: boolean;
  reminder_days: number[];
  last_generated_at?: string;
  next_generation_date?: string;
  created_at: string;
  updated_at: string;
}

export interface AutoInvoiceScheduleRequest {
  client_id: number;
  enabled: boolean;
  billing_day: number;
  auto_send_email: boolean;
  send_reminder: boolean;
  reminder_days: number[];
}

export interface SettlementApproval {
  settlement_id: number;
  status: string;
  approved_by?: number;
  approved_at?: string;
  approval_notes?: string;
}

export interface SettlementApprovalRequest {
  settlement_id: number;
  action: 'approve' | 'reject';
  notes?: string;
}

export interface SettlementApprovalHistory {
  id: number;
  settlement_id: number;
  action: string;
  actor_id?: number;
  actor_name?: string;
  notes?: string;
  created_at: string;
}

export interface BillingStatistics {
  period_start: string;
  period_end: string;
  total_invoices: number;
  total_amount: number;
  average_amount: number;
  by_status: Record<string, { count: number; amount: number }>;
  collection_efficiency: number;
}

export interface SettlementStatistics {
  period_start: string;
  period_end: string;
  total_settlements: number;
  total_amount: number;
  average_amount: number;
  approved_count: number;
  approval_rate: number;
}

export interface ExportRequest {
  invoice_ids?: number[];
  start_date?: string;
  end_date?: string;
  client_id?: number;
  status?: string;
  format: 'excel' | 'pdf';
  include_details: boolean;
}

export interface ExportTask {
  task_id: string;
  status: string;
  file_url?: string;
  created_at: string;
}

// ============= API Functions =============

/**
 * 요금 미리보기 - 실시간 요금 계산
 */
export const previewCharge = async (
  request: ChargePreviewRequest
): Promise<ChargePreviewResponse> => {
  const response = await apiClient.post<ChargePreviewResponse>(
    `${API_BASE_URL}/preview`,
    request
  );
  return response.data;
};

/**
 * 재무 대시보드 - 요약 정보
 */
export const getFinancialSummary = async (
  startDate?: string,
  endDate?: string
): Promise<FinancialSummary> => {
  const params: Record<string, string> = {};
  if (startDate) params.start_date = startDate;
  if (endDate) params.end_date = endDate;

  const response = await apiClient.get<FinancialSummary>(
    `${API_BASE_URL}/dashboard/financial`,
    { params }
  );
  return response.data;
};

/**
 * 월별 추이 조회
 */
export const getMonthlyTrends = async (
  startDate?: string,
  endDate?: string,
  months: number = 12
): Promise<MonthlyTrend[]> => {
  const params: Record<string, any> = { months };
  if (startDate) params.start_date = startDate;
  if (endDate) params.end_date = endDate;

  const response = await apiClient.get<MonthlyTrend[]>(
    `${API_BASE_URL}/dashboard/trends`,
    { params }
  );
  return response.data;
};

/**
 * 주요 고객 순위
 */
export const getTopClients = async (
  startDate?: string,
  endDate?: string,
  limit: number = 10
): Promise<TopClient[]> => {
  const params: Record<string, any> = { limit };
  if (startDate) params.start_date = startDate;
  if (endDate) params.end_date = endDate;

  const response = await apiClient.get<TopClient[]>(
    `${API_BASE_URL}/dashboard/top-clients`,
    { params }
  );
  return response.data;
};

/**
 * 재무 대시보드 데이터 조회 (getFinancialDashboard alias)
 */
export const getFinancialDashboard = getFinancialSummary;

/**
 * 자동 청구 스케줄 생성
 */
export const createAutoInvoiceSchedule = async (
  request: AutoInvoiceScheduleRequest
): Promise<AutoInvoiceSchedule> => {
  const response = await apiClient.post<AutoInvoiceSchedule>(
    `${API_BASE_URL}/auto-schedule`,
    request
  );
  return response.data;
};

/**
 * 자동 청구 스케줄 목록 조회
 */
export const getAutoInvoiceSchedules = async (
  enabled?: boolean
): Promise<AutoInvoiceSchedule[]> => {
  const params: Record<string, any> = {};
  if (enabled !== undefined) params.enabled = enabled;

  const response = await apiClient.get<AutoInvoiceSchedule[]>(
    `${API_BASE_URL}/auto-schedule`,
    { params }
  );
  return response.data;
};

/**
 * 고객별 자동 청구 스케줄 조회
 */
export const getAutoInvoiceSchedule = async (
  clientId: number
): Promise<AutoInvoiceSchedule> => {
  const response = await apiClient.get<AutoInvoiceSchedule>(
    `${API_BASE_URL}/auto-schedule/${clientId}`
  );
  return response.data;
};

/**
 * 자동 청구 스케줄 생성/수정
 */
export const createOrUpdateAutoInvoiceSchedule = async (
  request: AutoInvoiceScheduleRequest
): Promise<AutoInvoiceSchedule> => {
  const response = await apiClient.post<AutoInvoiceSchedule>(
    `${API_BASE_URL}/auto-schedule`,
    request
  );
  return response.data;
};

/**
 * 정산 승인/반려 처리
 */
export const processSettlementApproval = async (
  request: SettlementApprovalRequest
): Promise<SettlementApproval> => {
  const response = await apiClient.post<SettlementApproval>(
    `${API_BASE_URL}/settlement-approval`,
    request
  );
  return response.data;
};

/**
 * 정산 승인 상태 조회
 */
export const getSettlementApproval = async (
  settlementId: number
): Promise<SettlementApproval> => {
  const response = await apiClient.get<SettlementApproval>(
    `${API_BASE_URL}/settlement-approval/${settlementId}`
  );
  return response.data;
};

/**
 * 정산 승인 이력 조회
 */
export const getSettlementApprovalHistory = async (
  settlementId: number
): Promise<SettlementApprovalHistory[]> => {
  const response = await apiClient.get<{ settlement_id: number; history: SettlementApprovalHistory[] }>(
    `${API_BASE_URL}/settlement-approval/${settlementId}/history`
  );
  return response.data.history;
};

/**
 * 청구 통계
 */
export const getBillingStatistics = async (
  startDate?: string,
  endDate?: string
): Promise<BillingStatistics> => {
  const params: Record<string, string> = {};
  if (startDate) params.start_date = startDate;
  if (endDate) params.end_date = endDate;

  const response = await apiClient.get<BillingStatistics>(
    `${API_BASE_URL}/statistics/billing`,
    { params }
  );
  return response.data;
};

/**
 * 정산 통계
 */
export const getSettlementStatistics = async (
  startDate?: string,
  endDate?: string
): Promise<SettlementStatistics> => {
  const params: Record<string, string> = {};
  if (startDate) params.start_date = startDate;
  if (endDate) params.end_date = endDate;

  const response = await apiClient.get<SettlementStatistics>(
    `${API_BASE_URL}/statistics/settlement`,
    { params }
  );
  return response.data;
};

/**
 * 내보내기 작업 생성
 */
export const createExportTask = async (
  request: ExportRequest
): Promise<ExportTask> => {
  const response = await apiClient.post<ExportTask>(
    `${API_BASE_URL}/export`,
    request
  );
  return response.data;
};

/**
 * 내보내기 작업 상태 조회
 */
export const getExportTask = async (taskId: string): Promise<ExportTask> => {
  const response = await apiClient.get<ExportTask>(
    `${API_BASE_URL}/export/${taskId}`
  );
  return response.data;
};

// ============= Utility Functions =============

/**
 * 금액 포맷팅 (원화)
 */
export const formatCurrency = (amount: number): string => {
  return new Intl.NumberFormat('ko-KR', {
    style: 'currency',
    currency: 'KRW',
  }).format(amount);
};

/**
 * 퍼센트 포맷팅
 */
export const formatPercent = (value: number): string => {
  return `${value.toFixed(1)}%`;
};

/**
 * 날짜 포맷팅
 */
export const formatDate = (date: string): string => {
  return new Date(date).toLocaleDateString('ko-KR', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });
};

/**
 * 기간 텍스트 생성
 */
export const getPeriodText = (startDate: string, endDate: string): string => {
  const start = new Date(startDate);
  const end = new Date(endDate);
  
  if (start.getFullYear() === end.getFullYear() && start.getMonth() === end.getMonth()) {
    return start.toLocaleDateString('ko-KR', {
      year: 'numeric',
      month: 'long',
    });
  }
  
  return `${formatDate(startDate)} ~ ${formatDate(endDate)}`;
};

/**
 * 이번 달 시작일/종료일
 */
export const getCurrentMonthDates = (): { startDate: string; endDate: string } => {
  const now = new Date();
  const startDate = new Date(now.getFullYear(), now.getMonth(), 1);
  const endDate = new Date(now.getFullYear(), now.getMonth() + 1, 0);
  
  return {
    startDate: startDate.toISOString().split('T')[0],
    endDate: endDate.toISOString().split('T')[0],
  };
};

/**
 * 지난 달 시작일/종료일
 */
export const getLastMonthDates = (): { startDate: string; endDate: string } => {
  const now = new Date();
  const startDate = new Date(now.getFullYear(), now.getMonth() - 1, 1);
  const endDate = new Date(now.getFullYear(), now.getMonth(), 0);
  
  return {
    startDate: startDate.toISOString().split('T')[0],
    endDate: endDate.toISOString().split('T')[0],
  };
};

/**
 * 최근 N개월 시작일/종료일
 */
export const getLastNMonthsDates = (months: number): { startDate: string; endDate: string } => {
  const now = new Date();
  const endDate = new Date(now.getFullYear(), now.getMonth() + 1, 0);
  const startDate = new Date(now.getFullYear(), now.getMonth() - months + 1, 1);
  
  return {
    startDate: startDate.toISOString().split('T')[0],
    endDate: endDate.toISOString().split('T')[0],
  };
};

// ============= Additional Helper Functions =============

/**
 * Execute due auto invoices
 */
export const executeAutoInvoices = async (): Promise<{ scheduled: number; executed: number; failed: number }> => {
  const response = await apiClient.post<{ scheduled: number; executed: number; failed: number }>(
    `${API_BASE_URL}/auto-schedule/execute-due`
  );
  return response.data;
};

/**
 * Get settlement approvals list
 */
export const getSettlementApprovals = async (
  status?: string
): Promise<SettlementApproval[]> => {
  const params: Record<string, any> = {};
  if (status) params.status = status;

  const response = await apiClient.get<SettlementApproval[]>(
    `${API_BASE_URL}/settlement-approval`,
    { params }
  );
  return response.data;
};

/**
 * Approve settlement
 */
export const approveSettlement = async (
  settlementId: number, 
  comments: string
): Promise<SettlementApproval> => {
  const response = await apiClient.post<SettlementApproval>(
    `${API_BASE_URL}/settlement-approval/${settlementId}/approve`,
    { comments }
  );
  return response.data;
};

/**
 * Reject settlement
 */
export const rejectSettlement = async (
  settlementId: number, 
  comments: string
): Promise<SettlementApproval> => {
  const response = await apiClient.post<SettlementApproval>(
    `${API_BASE_URL}/settlement-approval/${settlementId}/reject`,
    { comments }
  );
  return response.data;
};

/**
 * Get payment reminders
 */
export const getPaymentReminders = async (
  status?: string
): Promise<any[]> => {
  const params: Record<string, any> = {};
  if (status) params.status = status;

  const response = await apiClient.get<any[]>(
    `${API_BASE_URL}/payment-reminder`,
    { params }
  );
  return response.data;
};

/**
 * Create payment reminder
 */
export const createPaymentReminder = async (data: any): Promise<any> => {
  const response = await apiClient.post(`${API_BASE_URL}/payment-reminder`, data);
  return response.data;
};

/**
 * Send due payment reminders
 */
export const sendDuePaymentReminders = async (): Promise<{ sent: number; failed: number }> => {
  const response = await apiClient.post<{ sent: number; failed: number }>(
    `${API_BASE_URL}/payment-reminder/send-due`
  );
  return response.data;
};

/**
 * Delete payment reminder
 */
export const deletePaymentReminder = async (reminderId: number): Promise<void> => {
  await apiClient.delete(`${API_BASE_URL}/payment-reminder/${reminderId}`);
};

/**
 * Get export tasks list
 */
export const getExportTasks = async (
  status?: string
): Promise<ExportTask[]> => {
  const params: Record<string, any> = {};
  if (status) params.status = status;

  const response = await apiClient.get<ExportTask[]>(
    `${API_BASE_URL}/export`,
    { params }
  );
  return response.data;
};

/**
 * Update or create auto invoice schedule
 */
export const updateAutoInvoiceSchedule = async (
  scheduleId: number, 
  data: Partial<AutoInvoiceScheduleRequest>
): Promise<AutoInvoiceSchedule> => {
  const response = await apiClient.put<AutoInvoiceSchedule>(
    `${API_BASE_URL}/auto-schedule/${scheduleId}`,
    data
  );
  return response.data;
};

/**
 * Delete auto invoice schedule
 */
export const deleteAutoInvoiceSchedule = async (scheduleId: number): Promise<void> => {
  await apiClient.delete(`${API_BASE_URL}/auto-schedule/${scheduleId}`);
};

File: backend/app/schemas/billing_enhanced.py
"""
Phase 8: 결제/정산 시스템 강화 - 스키마
"""
from pydantic import BaseModel, Field, validator
from typing import Optional, List, Dict, Any
from datetime import date, datetime
from decimal import Decimal


# ============= 요금 미리보기 =============

class ChargePreviewRequest(BaseModel):
    """요금 미리보기 요청"""
    client_id: int = Field(..., description="고객 ID")
    dispatch_date: date = Field(..., description="배차 날짜")
    total_distance_km: float = Field(..., ge=0, description="총 거리 (km)")
    pallets: int = Field(0, ge=0, description="팔레트 수")
    weight_kg: Optional[float] = Field(None, ge=0, description="무게 (kg)")
    vehicle_type: Optional[str] = Field(None, description="차량 타입 (냉동/냉장)")
    is_urgent: bool = Field(False, description="긴급 배차 여부")
    
    @validator('dispatch_date')
    def validate_dispatch_date(cls, v):
        if v < date.today():
            raise ValueError('배차 날짜는 오늘 이후여야 합니다')
        return v


class ChargeBreakdown(BaseModel):
    """요금 상세 내역"""
    base_distance_charge: float = Field(..., description="거리 기본 요금")
    base_pallet_charge: float = Field(..., description="팔레트 기본 요금")
    base_weight_charge: float = Field(0.0, description="무게 기본 요금")
    subtotal: float = Field(..., description="소계")
    
    # 할증
    weekend_surcharge: float = Field(0.0, description="주말 할증")
    night_surcharge: float = Field(0.0, description="야간 할증")
    express_surcharge: float = Field(0.0, description="긴급 할증")
    temperature_control_charge: float = Field(0.0, description="온도 관리 요금")
    total_surcharge: float = Field(..., description="총 할증")
    
    # 할인
    volume_discount: float = Field(0.0, description="물량 할인")
    special_discount: float = Field(0.0, description="특별 할인")
    total_discount: float = Field(..., description="총 할인")
    
    # 최종
    subtotal_after_discount: float = Field(..., description="할인 후 소계")
    tax_amount: float = Field(..., description="부가세 (10%)")
    total_amount: float = Field(..., description="최종 금액")


class ChargePreviewResponse(BaseModel):
    """요금 미리보기 응답"""
    client_id: int
    dispatch_date: date
    breakdown: ChargeBreakdown
    policy_info: Dict[str, Any] = Field(..., description="적용된 청구 정책 정보")
    notes: List[str] = Field(default_factory=list, description="안내 사항")


# ============= 자동 청구서 생성 =============

class AutoInvoiceScheduleCreate(BaseModel):
    """자동 청구서 생성 스케줄 설정"""
    client_id: int = Field(..., description="고객 ID")
    enabled: bool = Field(True, description="스케줄 활성화 여부")
    billing_day: int = Field(..., ge=1, le=28, description="청구일 (1-28)")
    auto_send_email: bool = Field(False, description="이메일 자동 발송")
    send_reminder: bool = Field(True, description="결제 알림 발송")
    reminder_days: List[int] = Field([7, 3, 0], description="알림 발송일 (D-day)")


class AutoInvoiceScheduleResponse(BaseModel):
    """자동 청구서 생성 스케줄 응답"""
    id: int
    client_id: int
    enabled: bool
    billing_day: int
    auto_send_email: bool
    send_reminder: bool
    reminder_days: List[int]
    last_generated_at: Optional[datetime]
    next_generation_date: Optional[date]
    created_at: datetime
    updated_at: datetime


# ============= 정산 승인 워크플로우 =============

class SettlementApprovalRequest(BaseModel):
    """정산 승인/반려 요청"""
    settlement_id: int = Field(..., description="정산서 ID")
    action: str = Field(..., description="승인 액션 (approve/reject)")
    notes: Optional[str] = Field(None, description="승인/반려 사유")
    
    @validator('action')
    def validate_action(cls, v):
        if v not in ['approve', 'reject']:
            raise ValueError('action은 approve 또는 reject만 가능합니다')
        return v


class SettlementApprovalResponse(BaseModel):
    """정산 승인 응답"""
    settlement_id: int
    status: str
    approved_by: Optional[int]
    approved_at: Optional[datetime]
    approval_notes: Optional[str]


class SettlementApprovalHistory(BaseModel):
    """정산 승인 이력"""
    id: int
    settlement_id: int
    action: str  # created, submitted, approved, rejected, paid
    actor_id: Optional[int]
    actor_name: Optional[str]
    notes: Optional[str]
    created_at: datetime


# ============= 전자세금계산서 =============

class TaxInvoiceIssueRequest(BaseModel):
    """전자세금계산서 발행 요청"""
    invoice_id: int = Field(..., description="청구서 ID")
    supplier_registration_number: str = Field(..., description="공급자 사업자등록번호")
    supplier_company_name: str = Field(..., description="공급자 상호")
    supplier_ceo_name: str = Field(..., description="공급자 대표자명")
    supplier_address: str = Field(..., description="공급자 주소")
    
    buyer_registration_number: str = Field(..., description="공급받는자 사업자등록번호")
    buyer_company_name: str = Field(..., description="공급받는자 상호")
    buyer_ceo_name: str = Field(..., description="공급받는자 대표자명")
    buyer_address: str = Field(..., description="공급받는자 주소")
    buyer_email: str = Field(..., description="공급받는자 이메일")
    
    issue_type: str = Field("정발행", description="발행 유형 (정발행/역발행)")
    remarks: Optional[str] = Field(None, description="비고")


class TaxInvoiceResponse(BaseModel):
    """전자세금계산서 응답"""
    id: int
    invoice_id: int
    tax_invoice_number: str
    issue_date: date
    supply_amount: float  # 공급가액
    tax_amount: float  # 세액
    total_amount: float  # 합계
    status: str  # pending, issued, cancelled
    nts_confirm_number: Optional[str]  # 국세청 승인번호
    issued_at: Optional[datetime]
    cancelled_at: Optional[datetime]


# ============= 결제 알림 =============

class PaymentReminderCreate(BaseModel):
    """결제 알림 생성"""
    invoice_id: int = Field(..., description="청구서 ID")
    reminder_type: str = Field(..., description="알림 유형 (before/due/overdue)")
    days_until_due: Optional[int] = Field(None, description="결제일까지 남은 일수")
    channels: List[str] = Field(["email"], description="알림 채널 (email/sms/push)")


class PaymentReminderResponse(BaseModel):
    """결제 알림 응답"""
    id: int
    invoice_id: int
    reminder_type: str
    sent_at: datetime
    channels: List[str]
    status: str  # sent, failed, pending


# ============= 재무 대시보드 =============

class FinancialSummaryResponse(BaseModel):
    """재무 요약 대시보드"""
    period_start: date
    period_end: date
    
    # 매출
    total_revenue: float = Field(..., description="총 매출")
    invoiced_amount: float = Field(..., description="청구 금액")
    collected_amount: float = Field(..., description="수금 금액")
    collection_rate: float = Field(..., description="수금률 (%)")
    
    # 미수금
    total_receivables: float = Field(..., description="총 미수금")
    current_receivables: float = Field(..., description="정상 미수금")
    overdue_receivables: float = Field(..., description="연체 미수금")
    overdue_count: int = Field(..., description="연체 건수")
    
    # 정산
    total_settlements: float = Field(..., description="총 정산 금액")
    pending_settlements: float = Field(..., description="정산 대기 금액")
    paid_settlements: float = Field(..., description="지급 완료 금액")
    
    # 현금 흐름
    cash_in: float = Field(..., description="현금 유입")
    cash_out: float = Field(..., description="현금 유출")
    net_cash_flow: float = Field(..., description="순 현금 흐름")


class MonthlyTrendResponse(BaseModel):
    """월별 추이"""
    month: str  # YYYY-MM
    revenue: float
    collected: float
    settlements: float
    net_profit: float


class TopClientResponse(BaseModel):
    """주요 고객 매출"""
    client_id: int
    client_name: str
    total_revenue: float
    total_collected: float
    invoice_count: int
    collection_rate: float


# ============= Excel/PDF 내보내기 =============

class InvoiceExportRequest(BaseModel):
    """청구서 내보내기 요청"""
    invoice_ids: Optional[List[int]] = Field(None, description="청구서 ID 목록 (없으면 전체)")
    start_date: Optional[date] = Field(None, description="시작 날짜")
    end_date: Optional[date] = Field(None, description="종료 날짜")
    client_id: Optional[int] = Field(None, description="특정 고객")
    status: Optional[str] = Field(None, description="청구서 상태")
    format: str = Field("excel", description="내보내기 형식 (excel/pdf)")
    include_details: bool = Field(True, description="상세 항목 포함 여부")


class ExportResponse(BaseModel):
    """내보내기 응답"""
    task_id: str = Field(..., description="작업 ID")
    status: str = Field(..., description="작업 상태")
    file_url: Optional[str] = Field(None, description="다운로드 URL")
    created_at: datetime


# ============= 청구서 상세 조회 향상 =============

class InvoiceDetailResponse(BaseModel):
    """청구서 상세 응답 (향상)"""
    id: int
    invoice_number: str
    client_id: int
    client_name: str
    
    # 청구 기간
    billing_period_start: date
    billing_period_end: date
    issue_date: date
    due_date: date
    
    # 금액
    subtotal: float
    tax_amount: float
    discount_amount: float
    total_amount: float
    paid_amount: float
    balance: float
    
    # 상태
    status: str
    is_overdue: bool
    days_overdue: Optional[int]
    
    # 항목
    line_items: List[Dict[str, Any]]
    
    # 결제 이력
    payment_history: List[Dict[str, Any]]
    
    # 전자세금계산서
    tax_invoice: Optional[TaxInvoiceResponse]
    
    # 메타
    created_at: datetime
    updated_at: datetime
    notes: Optional[str]


class SettlementDetailResponse(BaseModel):
    """정산서 상세 응답 (향상)"""
    id: int
    settlement_number: str
    driver_id: int
    driver_name: str
    
    # 정산 기간
    period_start: date
    period_end: date
    
    # 금액
    total_amount: float
    paid_amount: float
    balance: float
    
    # 상태
    status: str
    approval_status: str  # pending, approved, rejected
    
    # 항목
    line_items: List[Dict[str, Any]]
    
    # 승인 이력
    approval_history: List[SettlementApprovalHistory]
    
    # 메타
    created_at: datetime
    approved_at: Optional[datetime]
    paid_at: Optional[datetime]
    notes: Optional[str]


# ============= 통계 =============

class BillingStatisticsResponse(BaseModel):
    """청구 통계"""
    period_start: date
    period_end: date
    
    total_invoices: int
    total_amount: float
    average_amount: float
    
    by_status: Dict[str, Dict[str, Any]]  # {status: {count, amount}}
    by_client: List[TopClientResponse]
    
    payment_methods: Dict[str, int]  # {method: count}
    collection_efficiency: float  # 수금 효율


class SettlementStatisticsResponse(BaseModel):
    """정산 통계"""
    period_start: date
    period_end: date
    
    total_settlements: int
    total_amount: float
    average_amount: float
    
    by_status: Dict[str, Dict[str, Any]]
    by_driver: List[Dict[str, Any]]
    
    approval_rate: float
    average_approval_days: float
